bench/utils/__init__.py

```diff
@@ -50,6 +50,7 @@ def is_frappe_app(directory: str) -> bool:
 	return bool(is_frappe_app)
 
 
+@lru_cache(maxsize=None)
```

Comment:
PS: cache is being used here so that when the same function is called multiple times in the same process, it doesn't need to ask for git creds multiple times if the the git url uses http(s)